### DISCUSSION TITLE
Accessing classes and methods from DLL.

### DISCUSSION THREAD

**Discussion by SheldonPatnett** [pythonnet/pythonnet#2219] [Q&A]
### Environment

-   Pythonnet version: 3.0.1
-   Python version: Python 3.11.3
-   Operating System: ArchLinux
-   .NET Runtime: dotnet --info
.NET SDK:
 Version:   7.0.109
 Commit:    3e9283a8e9

Runtime Environment:
 OS Name:     ArchLinux
 OS Version:  
 OS Platform: Linux
 RID:         arch-x64
 Base Path:   /usr/share/dotnet/sdk/7.0.109/

Host:
  Version:      7.0.9
  Architecture: x64
  Commit:       8e9a17b221

.NET SDKs installed:
  6.0.120 [/usr/share/dotnet/sdk]
  7.0.109 [/usr/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.NETCore.App 6.0.20 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.9 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  DOTNET_ROOT       [/usr/share/dotnet]

### Details

-   Describe what you were trying to get done.
I'm trying to use Photon3Unity3D.dll to parse Photon packets I've captured https://github.com/reznok/ROAMapper/blob/master/ROAMapper/bin/Release/lib/Photon3Unity3D.dll
    _TODO_

-   What commands did you run to trigger this issue? If you can provide a
    [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
    this will help us understand the issue.

I've found something I believe parses photonpackets inside the dll using ILSpy https://i.imgur.com/0EKDI8u.png

`python
import clr

# Load the Unity3D Photon assembly
clr.AddReference("/home/m6/Projects/Assist/Photon3Unity3D.dll")
import ExitGames.Client.Photon

# Import the SocketUdp class
from ExitGames.Client.Photon import SocketUdp

# Create an instance of the SocketUdp class
socket = SocketUdp()
`

```python
    print('TODO')
```

-   If there was a crash, please include the traceback here.

```python
Traceback (most recent call last):
  File "/home/m6/Projects/Assist/test.py", line 8, in <module>
    from ExitGames.Client.Photon import SocketUdp
ImportError: cannot import name 'SocketUdp' from 'ExitGames.Client.Photon' (unknown location)
```

I'm sorry if this issue isn't properly formatted as I'm not a Windows user or .NET developer. Any help or guidance in the right direction would be greatly appreciated.

**Comment by sb5m** [comment]
Hi @SheldonPatnett, are the classes and methods public?
